### PR TITLE
Adds Create, Read and Delete for Systems with Baselines

### PR DESCRIPTION
**Pull request by petrblaho**
https://issues.redhat.com/browse/DRFT-249
https://issues.redhat.com/browse/DRFT-250
https://issues.redhat.com/browse/DRFT-251

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [x] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by eherget** [comment]
@petrblaho Got some unit test, flake8 and black issues to address.  Once we get this PR merged, I'll wire in the db stuff in my PR will the necessary queries.

File: system_baseline/openapi/api.spec.yaml [hunk 1]
```diff
@@ -201,6 +201,96 @@ paths:
           $ref: '#/components/responses/Forbidden'
         '500':
           $ref: '#/components/responses/InternalServerError'
+  /baselines/{baseline_id}/systems:
```
Comment (CodeHeeler) [inline system_baseline/openapi/api.spec.yaml:204]: This is great for the portion needed for frontend for the new feature.  What about a way for hsps to perform the query of what baselines are associated with one given system id?  If that info is still stored only in the baselines db, then we still need an api endpoint for hsp to call and my understanding from beav is that it needs to be much higher, like at the same level as /mgmt/, we'll add an /internal/.  Should that be a separate PR?  If so, I'll try to work on that, just let me know.

Comment (petrblaho) [reply system_baseline/openapi/api.spec.yaml:204]: That will most probably will be different PR.

Also is this the way we want to go?
I thought that we will use that shared secret approach and the API endpoint will not be internal.

File: system_baseline/openapi/api.spec.yaml [hunk 2]
```diff
@@ -342,6 +432,26 @@ components:
             type: string
             minLength: 32
             maxLength: 36
+    System:
+      type: object
```
[the remark this answers is not in the document]
Comment (petrblaho) [reply system_baseline/openapi/api.spec.yaml:421]: You are right, thank you

**Comment by petrblaho** [comment]
This code uses class and methods from https://github.com/RedHatInsights/system-baseline-backend/pull/157

File: system_baseline/openapi/api.spec.yaml (hunk 1 above)
Comment (CodeHeeler) [reply system_baseline/openapi/api.spec.yaml:204]: I think it's sort of both to keep people from attempting to break it?  Either way, as you say, we'll discuss it on its own PR. :)

**Comment by petrblaho** [comment]
@CodeHeeler I made some minor changes. Can you please review this code again?

**Review by CodeHeeler — APPROVED** [review]
LGTM, thanks!

**Comment by msager27** [comment]
It looks like we're putting the verb in the endpoint.  Maybe we can use the http verb instead like we do with other baseline operations:

POST is create
GET is read
DELETE is delete
PATCH is edit

**Review by msager27 — COMMENTED** [review]
Elaborating on my comment, some proposed examples:

Read:
GET /baselines/{baseline_id}/systems (no body)

Create:
POST /baselines/{baseline_id}/systems (with system ids in the body)

Delete:
DELETE /baselines/{baseline_id}/systems/{list of system ids}  (reference how we do baseline deletes as we have two ways of doing them -- POST and DELETE)

Update:
PATCH /baselines/{baseline_id}/systems  (body would be a diff -- see baseline update)

**Comment by petrblaho** [comment]
Talked about the review notes with @msager27 and others and we decided to rewrite this API to mimic what we have with baselines API to have as much consistency as possible with the exception of PATCH method - which can be implemented later if needed (https://issues.redhat.com/browse/DRFT-286)

**Comment by petrblaho** [comment]
@johnsonm325 ping so you know about the API

**Review by msager27 — APPROVED** [review]

File: system_baseline/views/v1.py [hunk 3]
```diff
@@ -597,6 +597,84 @@ def update_baseline(baseline_id, system_baseline_patch):
...
+    query = SystemBaseline.query.filter(
+        SystemBaseline.account == account_number, SystemBaseline.id == baseline_id
+    )
+    baseline = query.first_or_404()
+
+    message = "read baseline"
+    current_app.logger.audit(message, request=request, success=True)
+
+    for system_id in system_ids:
+        baseline.add_mapped_system(system_id)
```
Comment (CodeHeeler) [inline system_baseline/views/v1.py:632]: Do we need try/except blocks here?

File: system_baseline/views/v1.py [hunk 4]
```diff
@@ -597,6 +597,84 @@ def update_baseline(baseline_id, system_baseline_patch):
...
+    query = SystemBaseline.query.filter(
+        SystemBaseline.account == account_number, SystemBaseline.id == baseline_id
+    )
+    baseline = query.first_or_404()
+
+    message = "read baseline"
+    current_app.logger.audit(message, request=request, success=True)
+
+    for system_id in system_ids:
+        baseline.remove_mapped_system(system_id)
```
Comment (CodeHeeler) [inline system_baseline/views/v1.py:658]: Do we need try/except blocks here?

**Review by CodeHeeler — APPROVED** [review]
LGTM, thanks for all the work!

**Pull request merged**